scripts/algorithms/fm.dml

```diff
@@ -0,0 +1,80 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+# Implements the (2-way) factorization machines.
+#
+# ------------------------------------------------------------------------------------------------------
+# NAME      TYPE     DEFAULT  MEANING
+# ------------------------------------------------------------------------------------------------------
+# X         String   ---      Location to read the input matrix X to be factorized
+# y         String   ---      Location to read the input matrix y corresponding to the feature vector
+# k         Int      ---      hyperparameter that defines the dimensionality of factorization
+# -------------------------------------------------------------------------------------------------------
+#
+# OUTPUT: Updated Parameters w0, W, V, in matrix form. 
+# 
+
+fileX = $X; # nxm : n rows and m columns
```

Comment:
Therefore, we could use:

```
X = (N, n) -> N examples with n features
w_0 = (1,) -> bias
W = (n, 1) -> strength of each feature
V = (n, k) -> factorized interaction terms
```

Or, rename the shapes to `n` examples and `d` features as:

```
X = (n, d) -> n examples with d features
w_0 = (1,) -> bias
W = (d, 1) -> strength of each feature
V = (d, k) -> factorized interaction terms
```

I think the latter with a note would be easier to quickly understand since it is more consistent with other algorithms.